File: practice3/FirebaseListExample/app/src/main/java/com/jh/firebaselistexample/MainActivity.java
package com.jh.firebaselistexample;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.util.Log;

import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private RecyclerView.Adapter adapter;
    private RecyclerView.LayoutManager layoutManager;
    private ArrayList<User>arrayList;
    // mainactivity는 firebase 와 adapter사이를 연결해주는 중간다리 역할을 한다
    private FirebaseDatabase database;
    private DatabaseReference databaseReference;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //firebase 와 연결을 시켜줘야 firebase 구문을 사용할 수 있다

        recyclerView=findViewById(R.id.recyclerView);// recyclerView연결
        recyclerView.setHasFixedSize(true); //리사이클러뷰 기존성능 강화
        layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);
        arrayList = new ArrayList<>(); // user 객체를 담을 어레이 리스트(어뎁터 쪽으로

        database = FirebaseDatabase.getInstance();// 파이에 베이스 데이터베이스 연동

        databaseReference = database.getReference("User");
        //DB 데이터 연결
        // user는 파이어 베이스 db의 데이터 베이스명
        databaseReference.addListenerForSingleValueEvent(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                // 파이어베이스 데이터베이스의 데이터를 받아오는 곳
                arrayList.clear(); // 기존 배열 초기화
                for(DataSnapshot snapshot : dataSnapshot.getChildren()){ // 반복문으로 데이터 List를 추출해냄
                    User user = snapshot.getValue(User.class); // 만들어뒀던 User객체에 데이터를 담는다.
                    arrayList.add(user); //담은 데이터들을 배열리스트에 넣고 리사이클러뷰로 보낼 준비
                }
                adapter.notifyDataSetChanged(); // 리스트 저장 및 새로고침
            }

            @Override
            public void onCancelled(@NonNull DatabaseError databaseError) {
                // db를 가져오던 중 에러 발생 시
                Log.e("MainActivity", String.valueOf(databaseError.toException()));// 에러 발생시 출력
            }
        });// end databaseReference

        adapter= new CustomAdapter(arrayList, this);
        recyclerView.setAdapter(adapter); // 리사이클러뷰에 어댑터 연결


    }
}

File: practice3/hongdroid10_recyclerviewkt/src/main/java/com/jh/hongdroid10_recyclerviewkt/MainActivity.kt
package com.jh.hongdroid10_recyclerviewkt

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val rv_profile: RecyclerView= findViewById(R.id.rv_profile)

        val profileList = arrayListOf(
            Profiles(R.drawable.man, "홍드로이드",27,"안드로이드 앱 개발자"), //Profiles.kt 의 들어가야 될 값을 순서대로 집어 넣는다!
            Profiles(R.drawable.woman, "김드로이드",28,"안드로이드 앱 개발자"),
            Profiles(R.drawable.man, "이드로이드",29,"안드로이드 앱 개발자"),
            Profiles(R.drawable.woman, "박드로이드",30,"안드로이드 앱 개발자"),
            Profiles(R.drawable.man, "홍드로이드",27,"안드로이드 앱 개발자"),
            Profiles(R.drawable.woman, "김드로이드",28,"안드로이드 앱 개발자"),
            Profiles(R.drawable.man, "이드로이드",29,"안드로이드 앱 개발자"),
            Profiles(R.drawable.woman, "박드로이드",30,"안드로이드 앱 개발자")

        )
        //this -> mainActivity가 가지고 있는 정보(값)들
        rv_profile.layoutManager=LinearLayoutManager(this, LinearLayoutManager.VERTICAL,false)
        rv_profile.setHasFixedSize(true)

        rv_profile.adapter =ProfileAdapter(profileList) // 만들어놨던 ProfileAdapter 최초 생성!
    }
}